activesupport/lib/active_support/current_attributes.rb

```diff
@@ -0,0 +1,170 @@
+module ActiveSupport
+  # Abstract super class that provides a thread-isolated attributes singleton. 
+  # Primary use case is keeping all the per-request attributes easily available to the whole system.
+  #
+  # The following full app-like example demonstrates how to use a Current class to
+  # facilitate easy access to the global, per-request attributes without passing them deeply
+  # around everywhere:
+  #
+  #   # app/services/current.rb
+  #   require 'active_support/current_attributes'
```

Comment:
I just set `CurrentAttributes` up as an autoload in the top level `active_support.rb`.